<a href="https://colab.research.google.com/github/awang1020/Python-project/blob/main/Iris%20AutoML/PyCaret_Tuto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Introduction**

Bonjour à tous.
Nous allons utiliser une librairie appelée Pycaret sur Python.
Pycaret est simple et facile à apprendre, est une librairie low-code de machine learning avec Python, qui automatise des tâches de ML.

Nous allons voir comment intégrer cette librairie dans un contexte de machine learning classification multiclasse.



# **1.1 Installation de PyCaret**

Utilisation de la commannde 
!pip install pycaret





In [ ]:
!pip install pycaret

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 320 kB 4.2 MB/s 
     |████████████████████████████████| 3.3 MB 49.4 MB/s 
     |████████████████████████████████| 88 kB 7.2 MB/s 
     |████████████████████████████████| 1.7 MB 34.2 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 167 kB 44.2 MB/s 
     |████████████████████████████████| 6.8 MB 36.2 MB/s 
     |████████████████████████████████| 56 kB 4.1 MB/s 
     |████████████████████████████████| 4.8 MB 52.1 MB/s 
     |████████████████████████████████| 315 kB 71.3 MB/s 
     |████████████████████████████████| 17.0 MB 59.3 MB/s 
     |████████████████████████████████| 141 kB 67.6 MB/s 
     |████████████████████████████████| 2.0 MB 35.0 MB/s 
     |████████████████████████████████| 

**1.2 Pré-Requis**

Prérequis de version : 
Python 3.6 ou supérieur

PyCaret 2.0 ou supérieur



In [ ]:
from pycaret.utils import version
version()

'2.3.10'

# **2. Données**

Pour ce tutorial, nous allons utiliser Iris dataset. c’est un jeu de données qui contient 3 classes avec 50 enregistrements de chaque classe, et une classe désigne un type de la plante d’Iris.

Nous verrons dans un premier temps l’implémentation d’un modèle de machine learning classique puis nous allons voir comment l’autoML permet de nous donner un modèle et de tuner notre modèle.

On récupère les données grâce à la fonction **get_data()**.


In [ ]:
from pycaret.datasets import get_data
dataset = get_data('iris')

sepal_length  sepal_width  petal_length  petal_width      species
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

Ce jeu de données contient une description de la longueur et la largeur des sépales, la longueur et la largeur des pétales et le type d'Iris.

-----

Ensuite, pour réaliser la prédiction, nous allons déterminer une proportion de données pour faire l'entrainement et l'autre partie pour tester et réaliser la prédiction.

In [ ]:
data = dataset.sample(frac=0.9, random_state=786)
data_unseen = dataset.drop(data.index)

data.reset_index(drop=True, inplace=True)
data_unseen.reset_index(drop=True, inplace=True)

print('Données pour le modèle : ' + str(data.shape))
print('Données pour la prédiction: ' + str(data_unseen.shape))

Données pour le modèle : (135, 5)
Données pour la prédiction: (15, 5)


On a donc pris 90 % de notre jeu pour l'entrainement et 10% pour tester notre modèle.

# **3. Set up l'environnement PyCaret**

Maintenant, nous allons setup notre environnement dans PyCaret.

La fonction **setup()** est l'initialisation de l'environnement et crée des pipelines de transformations et préparations de données pour la modélisation et le déploiement.

Il prend en paramètre un dataframe pandas et une colonne cible.

Lorsque setup() est exécuté, **le type des variables sera déduit par PyCaret**, il nous demandera de valider le type en appuyant sur la touche "Entrée". Si ce n'est pas le cas, vous pouvez utiliser les paramètres numeric_features, categorical_features, date_features dans la fonction de configuration pour les spécifier.

S'assurer que les types de données sont corrects est d'une importance fondamentale dans PyCaret car il effectue automatiquement quelques tâches de prétraitement qui sont impératives pour toute expérience d'apprentissage automatique.


In [ ]:
from pycaret.classification import *
exp_mclf101 = setup(data = data, target = 'species', session_id=1) 

Description  \
0                               session_id   
1                                   Target   
2                              Target Type   
3                            Label Encoded   
4                            Original Data   
5                           Missing Values   
6                         Numeric Features   
7                     Categorical Features   
8                         Ordinal Features   
9                High Cardinality Features   
10                 High Cardinality Method   
11                   Transformed Train Set   
12                    Transformed Test Set   
13                      Shuffle Train-Test   
14                     Stratify Train-Test   
15                          Fold Generator   
16                             Fold Number   
17                                CPU Jobs   
18                                 Use GPU   
19                          Log Experiment   
20                         Experiment Name   
21                                     USI   
22                         Imputation Type   
23          Iterative Imputation Iteration   
24                         Numeric Imputer   
25      Iterative Imputation Numeric Model   
26                     Categorical Imputer   
27  Iterative Imputation Categorical Model   
28           Unknown Categoricals Handling   
29                               Normalize   
30                        Normalize Method   
31                          Transformation   
32                   Transformation Method   
33                                     PCA   
34                              PCA Method   
35                          PCA Components   
36                     Ignore Low Variance   
37                     Combine Rare Levels   
38                    Rare Level Threshold   
39                         Numeric Binning   
40                         Remove Outliers   
41                      Outliers Threshold   
42                Remove Multicollinearity   
43             Multicollinearity Threshold   
44             Remove Perfect Collinearity   
45                              Clustering   
46                    Clustering Iteration   
47                     Polynomial Features   
48                       Polynomial Degree   
49                    Trignometry Features   
50                    Polynomial Threshold   
51                          Group Features   
52                       Feature Selection   
53                Feature Selection Method   
54            Features Selection Threshold   
55                     Feature Interaction   
56                           Feature Ratio   
57                   Interaction Threshold   
58                           Fix Imbalance   
59                    Fix Imbalance Method   

                                                Value  
0                                                   1  
1                                             species  
2                                          Multiclass  
3   Iris-setosa: 0, Iris-versicolor: 1, Iris-virgi...  
4                                            (135, 5)  
5                                               False  
6                                                   4  
7                                                   0  
8                                               False  
9                                               False  
10                                               None  
11                                            (94, 4)  
12                                            (41, 4)  
13                                               True  
14                                              False  
15                                    StratifiedKFold  
16                                                 10  
17                                                 -1  
18                                              False  
19                                              False  
20                                   clf-default

INFO:logs:create_model_container: 0
INFO:logs:master_model_container: 0
INFO:logs:display_container: 1
INFO:logs:Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=True, features_todrop=[],
                                      id_columns=[],
                                      ml_usecase='classification',
                                      numerical_features=[], target='species',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='not_available',
                                fill_value_categorical=None,
                                fill_value_numerical=None,
                                numeric_stra...
                ('scaling', 'passthrough'), ('P_transform', 'passthrough'),
                ('binn', 'passthrough'), ('rem_outliers', 'passthrough'),
                ('cluste

*session_id* : Un nombre pseudo-aléatoire distribué comme seed dans toutes les fonctions pour une reproductibilité ultérieure. Ici, il est égal à 1.

# **4. Comparaison des modèles**

compare_models() est une fonction de PyCaret pour évaluer les performance de différents modèles, c'est le point de départ recommandé pour créer notre modèle.

La sortie imprime une grille de score qui affiche l'exactitude, le rappel, la précision, F1, Kappa et MCC par pli.

In [ ]:
best_model = compare_models(sort = 'Recall')

Model  Accuracy     AUC  Recall   Prec.  \
lda          Linear Discriminant Analysis    0.9900  1.0000  0.9889  0.9925   
qda       Quadratic Discriminant Analysis    0.9789  0.9963  0.9778  0.9842   
gbc          Gradient Boosting Classifier    0.9789  0.9890  0.9778  0.9837   
lr                    Logistic Regression    0.9689  1.0000  0.9667  0.9762   
rf               Random Forest Classifier    0.9689  0.9955  0.9667  0.9762   
knn                K Neighbors Classifier    0.9578  1.0000  0.9556  0.9678   
dt               Decision Tree Classifier    0.9578  0.9679  0.9556  0.9712   
lightgbm  Light Gradient Boosting Machine    0.9578  0.9971  0.9556  0.9678   
et                 Extra Trees Classifier    0.9489  0.9971  0.9444  0.9562   
nb                            Naive Bayes    0.9389  0.9877  0.9333  0.9508   
ada                  Ada Boost Classifier    0.9044  0.9783  0.9000  0.9295   
ridge                    Ridge Classifier    0.8211  0.0000  0.8111  0.8403   
svm                   SVM - Linear Kernel    0.7844  0.0000  0.7778  0.7176   
dummy                    Dummy Classifier    0.3500  0.5000  0.3333  0.1237   

              F1   Kappa     MCC  TT (Sec)  
lda       0.9897  0.9848  0.9862     0.022  
qda       0.9783  0.9682  0.9711     0.016  
gbc       0.9781  0.9679  0.9709     0.233  
lr        0.9678  0.9528  0.9571     0.608  
rf        0.9678  0.9528  0.9571     0.460  
knn       0.9564  0.9361  0.9420     0.116  
dt        0.9543  0.9361  0.9444     0.049  
lightgbm  0.9564  0.9361  0.9420     0.106  
et        0.9478  0.9225  0.9268     0.479  
nb        0.9353  0.9066  0.9145     0.015  
ada       0.8978  0.8558  0.8712     0.092  
ridge     0.8069  0.7294  0.7511     0.014  
svm       0.7260  0.6751  0.7326     0.073  
dummy     0.1824  0.0000  0.0000     0.013

INFO:logs:create_model_container: 14
INFO:logs:master_model_container: 14
INFO:logs:display_container: 2
INFO:logs:LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                           solver='svd', store_covariance=False, tol=0.0001)
INFO:logs:compare_models() succesfully completed......................................


Cette grille est essentiel pour choisir notre modèle.
Ici, par défaut, les modèles sont trié par l'accuracy, on peut très bien trier par Recall en faisant

compare_models(sort = 'Recall')

# **5. Création d'un modèle**

Prenons un exemple en créant un modèle.

create_model() est une fonction de PyCaret et permet de créer le modèle. Il est souvent à la base de la plupart des fonctionnalités de PyCaret. Comme son nom l'indique, cette fonction forme et évalue un modèle à l'aide de la cross validation qui peut être définie avec le paramètre fold. La sortie imprime une grille de score qui affiche l'exactitude, le recall, la précision, F1, Kappa et MCC par pli.

---
Prenons l'exemple de la régression logistique, avec un fold de 5, c'est à dire une validation croisée de 5 fois :

In [ ]:
lr = create_model('lr', fold = 5)

Accuracy     AUC  Recall   Prec.      F1   Kappa     MCC
Fold                                                          
0       0.8421  0.9831  0.8413  0.8447  0.8402  0.7615  0.7647
1       1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000
2       1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000
3       1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000
4       0.9444  1.0000  0.9444  0.9524  0.9441  0.9167  0.9209
Mean    0.9573  0.9966  0.9571  0.9594  0.9568  0.9356  0.9371
Std     0.0615  0.0067  0.0618  0.0602  0.0622  0.0929  0.0915

INFO:logs:create_model_container: 15
INFO:logs:master_model_container: 15
INFO:logs:display_container: 3
INFO:logs:LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)
INFO:logs:create_model() succesfully completed......................................


On voit qu'on a une **moyenne** de 0.9573 et un **écart type **de 0.0615.


# **6.0 Tuner un modèle**

Maintenant, essayons d'avoir une meilleur précision en utilisant les hyperparamètres.

Lorsqu'un modèle est créé à l'aide de la fonction create_model(), il utilise les hyperparamètres par défaut pour entraîner le modèle. 

Afin d'ajuster les hyperparamètres, la fonction **tune_model()** est utilisée.

 La sortie imprime une grille de score qui affiche la précision, l'AUC, le recall, la précision, la F1, le Kappa et le MCC pour le meilleur modèle.

In [ ]:
tuned_lr = tune_model(lr)

Accuracy  AUC  Recall  Prec.      F1   Kappa     MCC
Fold                                                      
0         0.90  1.0  0.8889  0.920  0.8956  0.8462  0.8598
1         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
2         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
3         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
4         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
5         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
6         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
7         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
8         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
9         1.00  1.0  1.0000  1.000  1.0000  1.0000  1.0000
Mean      0.99  1.0  0.9889  0.992  0.9896  0.9846  0.9860
Std       0.03  0.0  0.0333  0.024  0.0313  0.0462  0.0421

INFO:logs:create_model_container: 16
INFO:logs:master_model_container: 16
INFO:logs:display_container: 4
INFO:logs:LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)
INFO:logs:tune_model() succesfully completed......................................


Par défaut, il est réglé pour optimisé l'accuracy du modèle.

Les metrics (accuracy, recall...) seules ne sont pas les seuls critères à prendre en compte lors de la finalisation du meilleur modèle pour la production. D'autres facteurs à prendre en compte incluent le temps de formation, l'écart type des kfolds,...

# **7. Finaliser le modèle**

Enfin, finalisons le modèle

La finalisation du modèle est la dernière étape de l'expérience. 


La fonction **finalize_model()** adapte le modèle à l'ensemble de données complet, y compris sur l'échantillon test.


In [ ]:
final_lr = finalize_model(tuned_lr)

INFO:logs:Initializing finalize_model()
INFO:logs:finalize_model(estimator=LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False), fit_kwargs=None, groups=None, model_only=True, display=None, experiment_custom_tags=None, return_train_score=False)
INFO:logs:Finalizing LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)
INFO:logs:Initializing create_model()
INFO:logs:create_model(estimator=LogisticRegression(C=3.907, class_weight={}, dual=False, fit_interce

Affichons le modèle finale de machine learning multiclass pour ce jeu de données.

In [ ]:
print(final_lr)

LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)


# **8. Prédicition sur les données de test**

Appliquons ce modèle à nos données test qui contient 10 % de l'ensemble des données d'origine : nous allons utiliser la fonction **predict_model()**



In [ ]:
unseen_predictions = predict_model(final_lr, data=data_unseen)
unseen_predictions.head()

INFO:logs:Initializing predict_model()
INFO:logs:predict_model(estimator=LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False), probability_threshold=None, encoded_labels=False, drift_report=False, raw_score=False, round=4, verbose=True, ml_usecase=MLUsecase.CLASSIFICATION, display=None, drift_kwargs=None)
INFO:logs:Checking exceptions
INFO:logs:Preloading libraries
INFO:logs:Preparing display monitor


Model  Accuracy  AUC  Recall  Prec.  F1  Kappa  MCC
0  Logistic Regression       0.0  1.0       0      0   0      0    0

sepal_length  sepal_width  petal_length  petal_width          species  \
0           5.4          3.9           1.7          0.4      Iris-setosa   
1           5.4          3.4           1.7          0.2      Iris-setosa   
2           5.1          3.3           1.7          0.5      Iris-setosa   
3           4.8          3.1           1.6          0.2      Iris-setosa   
4           6.9          3.1           4.9          1.5  Iris-versicolor   

             Label   Score  
0      Iris-setosa  0.9868  
1      Iris-setosa  0.9702  
2      Iris-setosa  0.9753  
3      Iris-setosa  0.9821  
4  Iris-versicolor  0.8590

Les 2 dernières colonnes sont rajoutés : Label et Score.

Le label est la prédiction et le score est la probabilité de la prédiction.


# **9.0 Sauvegarde du modèle**

Nous avons maintenant terminé l'expérience en finalisant le modèle tuned_lr qui est maintenant stocké dans la variable final_lr. 

Nous avons également utilisé le modèle stocké dans final_lr pour prédire data_unseen. 

Ceci nous amène à la fin de notre expérience, mais une question reste à poser : que se passe-t-il lorsque vous avez de nouvelles données à prédire ? Devez-vous refaire toute l'expérience ? 

La réponse est non, la fonction intégrée save_model() de PyCaret vous permet d'enregistrer le modèle avec l'ensemble du pipeline de transformation pour une utilisation ultérieure.



In [ ]:
save_model(final_lr,'Final LR Model 19Nov2022')

INFO:logs:Initializing save_model()
INFO:logs:save_model(model=LogisticRegression(C=3.907, class_weight={}, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=1000,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=1, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False), model_name=Final LR Model 11Nov2022, prep_pipe_=Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=True, features_todrop=[],
                                      id_columns=[],
                                      ml_usecase='classification',
                                      numerical_features=[], target='species',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='not_available',
                   

Transformation Pipeline and Model Successfully Saved


(Pipeline(memory=None,
          steps=[('dtypes',
                  DataTypes_Auto_infer(categorical_features=[],
                                       display_types=True, features_todrop=[],
                                       id_columns=[],
                                       ml_usecase='classification',
                                       numerical_features=[], target='species',
                                       time_features=[])),
                 ('imputer',
                  Simple_Imputer(categorical_strategy='not_available',
                                 fill_value_categorical=None,
                                 fill_value_numerical=None,
                                 numeric_stra...
                 ('feature_select', 'passthrough'), ('fix_multi', 'passthrough'),
                 ('dfs', 'passthrough'), ('pca', 'passthrough'),
                 ['trained_model',
                  LogisticRegression(C=3.907, class_weight={}, dual=False,
                 

# **10.0 Charger le modèle sauvegardé**

Pour charger un modèle enregistré à une date ultérieure dans le même environnement ou dans un environnement alternatif, nous utiliserions la fonction **load_model()** de PyCaret, puis appliquerions facilement le modèle enregistré sur de nouvelles données pour la prédiction.

In [ ]:
saved_final_ln = load_model('Final LR Model 11Nov2022')

INFO:logs:Initializing load_model()
INFO:logs:load_model(model_name=Final LR Model 11Nov2022, platform=None, authentication=None, verbose=True)


Transformation Pipeline and Model Successfully Loaded


Une fois le modèle chargé dans l'environnement, vous pouvez simplement l'utiliser pour prédire de nouvelles données à l'aide de la même fonction predict_model().

In [ ]:
new_prediction = predict_model(saved_final_ln, data=data_unseen)
new_prediction.head()

INFO:logs:Initializing predict_model()
INFO:logs:predict_model(estimator=Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=True, features_todrop=[],
                                      id_columns=[],
                                      ml_usecase='classification',
                                      numerical_features=[], target='species',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='not_available',
                                fill_value_categorical=None,
                                fill_value_numerical=None,
                                numeric_stra...
                ('feature_select', 'passthrough'), ('fix_multi', 'passthrough'),
                ('dfs', 'passthrough'), ('pca', 'passthrough'),
                ['trained_model',
                 LogisticRegression

Model  Accuracy  AUC  Recall  Prec.  F1  Kappa  MCC
0  Logistic Regression       0.0  1.0       0      0   0      0    0

sepal_length  sepal_width  petal_length  petal_width          species  \
0           5.4          3.9           1.7          0.4      Iris-setosa   
1           5.4          3.4           1.7          0.2      Iris-setosa   
2           5.1          3.3           1.7          0.5      Iris-setosa   
3           4.8          3.1           1.6          0.2      Iris-setosa   
4           6.9          3.1           4.9          1.5  Iris-versicolor   

             Label   Score  
0      Iris-setosa  0.9868  
1      Iris-setosa  0.9702  
2      Iris-setosa  0.9753  
3      Iris-setosa  0.9821  
4  Iris-versicolor  0.8590

PyCaret est vraiment un bon outil d'autoML.

Nous avons vu qu'un workflow de machine learning dans PyCaret commence par setup(), suivi de la comparaison de tous les modèles à l'aide de compare_models() et du choix de modèle selon les critères de sélection (recall, accuracy...) pour ensuite, effectuer plusieurs techniques de modélisation telles que le réglage des hyperparamètres.

Tout cela est fait à partir d'un fonction facilement utilisable sous PyCaret.

Ce workflow vous mènera éventuellement au meilleur modèle à utiliser pour faire des prédictions sur des données nouvelles et inédites. 

# **11.0 La fonction automl**


Enfin, je vais vous présenter la fonction automl().

Cette fonction permet de sélectionner le meilleur modèle vu avec la fonction compare_models() basé sur le critère de sélection de votre choix. 

Par défaut, si aucun paramètre n'est renseigné, ce sera l'accuracy.

In [ ]:
best = automl()
best

INFO:logs:Initializing automl()
INFO:logs:automl(optimize=Accuracy, use_holdout=False, return_train_score=False)
INFO:logs:Model Selection Basis : CV Results on Training set
INFO:logs:Initializing create_model()
INFO:logs:create_model(estimator=LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                           solver='svd', store_covariance=False, tol=0.0001), fold=None, round=4, cross_validation=False, predict=False, fit_kwargs=None, groups=None, refit=True, verbose=False, system=False, metrics=None, experiment_custom_tags=None, add_to_model_list=True, probability_threshold=None, display=None, return_train_score=False, kwargs={})
INFO:logs:Checking exceptions
INFO:logs:Importing libraries
INFO:logs:Copying training dataset
INFO:logs:Defining folds
INFO:logs:Declaring metric variables
INFO:logs:Importing untrained model
INFO:logs:Declaring custom model
INFO:logs:Linear Discriminant Analysis Imported succesfully
INFO:logs:Cross validation set to False


LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                           solver='svd', store_covariance=False, tol=0.0001)

Faisons un print pour afficher le meilleur modèle avec les hyperparamètres.

In [ ]:
print(best)

LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                           solver='svd', store_covariance=False, tol=0.0001)


Faisons la prédiction sur les données test avec ce best_modèle : 

In [ ]:
predict_new = predict_model(best, data=data_unseen)
predict_new.head()

INFO:logs:Initializing predict_model()
INFO:logs:predict_model(estimator=LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
                           solver='svd', store_covariance=False, tol=0.0001), probability_threshold=None, encoded_labels=False, drift_report=False, raw_score=False, round=4, verbose=True, ml_usecase=MLUsecase.CLASSIFICATION, display=None, drift_kwargs=None)
INFO:logs:Checking exceptions
INFO:logs:Preloading libraries
INFO:logs:Preparing display monitor


Model  Accuracy  AUC  Recall  Prec.  F1  Kappa  MCC
0  Linear Discriminant Analysis       0.0  1.0       0      0   0      0    0

sepal_length  sepal_width  petal_length  petal_width          species  \
0           5.4          3.9           1.7          0.4      Iris-setosa   
1           5.4          3.4           1.7          0.2      Iris-setosa   
2           5.1          3.3           1.7          0.5      Iris-setosa   
3           4.8          3.1           1.6          0.2      Iris-setosa   
4           6.9          3.1           4.9          1.5  Iris-versicolor   

             Label   Score  
0      Iris-setosa  1.0000  
1      Iris-setosa  1.0000  
2      Iris-setosa  1.0000  
3      Iris-setosa  1.0000  
4  Iris-versicolor  0.9972

Le label est la prédiction et le score est la probabilité de la prédiction.

Nous voyons un score + élevé par rapport à notre modèle de régression logistique. 


Enfin, si nous voulons déployer notre modèle sur de nouvelles données, nous pouvons le sauvegarder et le déployer à nouveau (vu au point 9 et 10)